addons/account/static/src/js/section_and_note_fields_backend.js

```diff
@@ -1,106 +1,72 @@
+    }
 
-        return $cell;
-    },
-    /**
-     * We add the o_is_{display_type} class to allow custom behaviour both in JS and CSS.
-     *
-     * @override
-     */
-    _renderRow: function (record, index) {
-        var $row = this._super.apply(this, arguments);
+    getRowClass(record) {
+        const existingClasses = super.getRowClass(record);
+        return `${existingClasses} o_is_${record.data.display_type}`;
+    }
 
-        if (record.data.display_type) {
-            $row.addClass('o_is_' + record.data.display_type);
+    getCellClass(column, record) {
+        const classNames = super.getCellClass(column, record);
+        if (this.isSectionOrNote && column.widget !== "handle" && column.name !== "name") {
+            return `${classNames} o_hidden`;
         }
+        return classNames;
+    }
 
-        return $row;
-    },
-    /**
-     * We want to add .o_section_and_note_list_view on the table to have stronger CSS.
-     *
-     * @override
-     * @private
-     */
-    _renderView: function () {
-        var self = this;
-        return this._super.apply(this, arguments).then(function () {
-            self.$('.o_list_table').addClass('o_section_and_note_list_view');
-        });
+    getCellColspan(column) {
+        if (!this.isSectionOrNote || column.widget === "handle") {
+            return 1;
+        }
+        let nbrColumns = this.withHandleColumn ? this.state.columns.length + 1 : this.state.columns.length;
+        let withTrashColumn = this.props.activeActions && (this.props.activeActions.canDelete || this.props.activeActions.canUnlink || 0);
+        return nbrColumns - this.withHandleColumn - withTrashColumn;
     }
-});
+}
 
-// We create a custom widget because this is the cleanest way to do it:
-// to be sure this custom code will only impact selected fields having the widget
-// and not applied to any other existing ListRenderer.
-var SectionAndNoteFieldOne2Many = FieldOne2Many.extend({
-    /**
-     * We want to use our custom renderer for the list.
-     *
-     * @override
-     */
-    _getRenderer: function () {
-        if (this.view.arch.tag === 'tree') {
-            return SectionAndNoteListRenderer;
+export class SectionAndNoteFieldOne2Many extends X2ManyField {
+    setup() {
+        super.setup();
+        if (this.viewMode === 'list') {
+            this.Renderer = SectionAndNoteListRenderer;
         }
-        return this._super.apply(this, arguments);
-    },
-});
 
-// This is a merge between a FieldText and a FieldChar.
-// We want a FieldChar for section,
-// and a FieldText for the rest (product and note).
-var SectionAndNoteFieldText = function (parent, name, record, options) {
-    var isSection = record.data.display_type === 'line_section';
-    var Constructor = isSection ? FieldChar : ListFieldText;
-    return new Constructor(parent, name, record, options);
-};
+        const getList = () => this.list;
```

Comment:
I think there are better/cleaner ways to do this....
getting the list should not depend on section_and_note...